### ISSUE TITLE
FEATURE - Allow for multiple dbt sources

### ISSUE THREAD

**Issue by djbelknapdbs** [fivetran/dbt_salesforce_formula_utils#6]
Let me know if I missed something and this is possible now - this is based on my read of dbt and this package's documentation.

**Are you a Fivetran customer?** 
Yes

**Is your feature request related to a problem? Please describe.**
We currently sync Salesforce with multiple Fivetran connectors which means they end up in separate schema in Snowflake. Since dbt doesn't support multiple schemas per source, we have to create one source per connector (to my knowledge). This package assumes a single `snowflake` source.

**Describe the solution you'd like**
Include an option to point to a custom source/sources. This would ideally be a macro parameter so each run of the macro can specify which source the object is in.

**Describe alternatives you've considered**
Figure out a way to combine everything into a single schema before applying this package. That would ideally be done via dbt, but then this assumes a source() and not a ref().

**Please indicate the level of urgency and business impact of this request** 

Medium - overall we're trying to switch our production replication to Fivetran but need to get a few issues ironed out first.

**Are you interested in contributing to this package?** 

- [x] Yes, I can do this and open a PR for your review.
- [ ] Possibly, but I'm not quite sure how to do this. I'd be happy to do a live coding session with someone to get this work implemented. 
- [ ] No, I'd prefer if someone else did this.  I don't have the time and/or don't know how to incorporate the changes necessary.

I've got a fork going [here](https://github.com/djbelknapdbs/dbt_salesforce_formula_utils/tree/feature-source-name) that I can submit as a PR if you're interested in adding this feature.

**Comment by fivetran-joemarkiewicz** [comment]
Thank you for opening this Feature Request and providing a link to your fork!

I think this is a great idea and one we should include in a future release of the package. I just took a look at your fork and the logic makes sense to me and allows for a more dynamic ability for users to specific the source schema name if they do not want to use `salesforce`. You can submit your changes as a PR as this will be extremely helpful and is a great addition to the package.

Additionally, I have experienced customers in the past with a similar scenario of having multiple schemas and struggling to use a dbt package since it is limited to one target. What has been done in the past is create upstream models which union the tables from the schemas together, and then the you can point the dbt project to the unioned schema. There have been other alternatives to this. I will do a search for other alternatives and post back here with what I find.

Thank you!

**Comment by fivetran-joemarkiewicz** [comment]
@djbelknapdbs 

I have been able to create a draft PR that allows for users to override the name of the salesforce source! Unfortunately I was unable to get the solution you provided in your PR #9 to work on my end. I was receiving a runtime error when passing the argument into the source. The only instance I was able to get the source name changed was through the use of a variable set within the `dbt_project.yml`. This works great, however the limitation is that this source name MUST be applied to every macro used in your package and you cannot have various macros use different source names. 

Does this proposed solution conflict with how your project is set up? If it does, we can continue to search for a working solution. If this doesn't conflict, then you can go ahead and try the working branch with the below snippet and I can convert the PR for review ready!
```yml
##package.yml
packages:
  - git: https://github.com/fivetran/dbt_salesforce_formula_utils.git
    revision: feature/source-override
    warn-unpinned: false

```

**Comment by fivetran-joemarkiewicz** [comment]
@djbelknapdbs I wanted to check in if you were able to successfully use different sources with the branch mentioned above?

We are planning on merging and releasing this branch mid this week. Let us know if you had any issues. Thanks!

**Comment by djbelknapdbs** [comment]
Hi @fivetran-joemarkiewicz - sorry I didn't follow up - I didn't actually try that approach because it doesn't fit our use case. We have 3 schemas (sources) that all contain Salesforce objects, so we need it to be a macro-level setting rather than a project setting.

I think the changes in #4 conflict with my changes that happened in parallel. My approach with parameters should still work if it gets adjusted for the changes that came from that PR, if you're open to sticking with the macro parameters instead of the project variables. I can probably carve out some time to submit a new PR if that would help. The project variable probably works for most people, so if you'd rather stick with that I can also just handle this on my fork.

**Comment by fivetran-joemarkiewicz** [comment]
Thanks @djbelknapdbs that makes a lot of sense and I would agree that a maco approach would make the most sense for users going forward. If you do have time then I would be open to reviewing a PR without conflicts to accommodate the macro approach you originally took.

**Comment by fivetran-joemarkiewicz** [comment]
This Issue has been resolved in the most recent PR merge. You can expect this change to be reflected in the next release of this package. 

Closing the Issue, be sure to watch the releases of the package updates early next week!
